src/client/src/tscParseInsert.c

```diff
@@ -989,7 +989,7 @@ static int32_t tscCheckIfCreateTable(char **sqlstr, SSqlObj *pSql) {
 
 int validateTableName(char *tblName, int len) {
   char buf[TSDB_TABLE_ID_LEN] = {0};
-  strncpy(buf, tblName, len);
+  tstrncpy(buf, tblName, sizeof(buf));
```

Comment:
shall we redefine buf[TSDB_TABLE_ID_LEN] to buf[TSDB_TABLE_ID_LEN+1] ?
otherwise, it might conflict with the doc which says table id len is as TSDB_TABLE_ID_LEN.

Reply:
this patch is still work in progress,I create this PR to submit part of the changes avoid conflicts with other changes.
TSDB_TABLE_ID_LEN related issues will be fixed later.